uitools/cpp/Esri/ArcGISRuntime/Toolkit/OverviewMapController.cpp

```diff
@@ -0,0 +1,395 @@
+ *  limitations under the License.
+ ******************************************************************************/
+#include "OverviewMapController.h"
+
+// Toolkit headers
+#include "Internal/GeoViews.h"
+
+// ArcGISRuntime headers
+#include <Graphic.h>
+#include <GraphicsOverlay.h>
+#include <Map.h>
+#include <MapTypes.h>
+#include <SimpleFillSymbol.h>
+#include <SimpleLineSymbol.h>
+#include <SimpleMarkerSymbol.h>
+#include <Viewpoint.h>
+
+namespace Esri {
+namespace ArcGISRuntime {
+namespace Toolkit {
+
+  namespace {
+
+    /*!
+      \internal
+      Connect to MapView's viewpointChanged. Note, we display the visible area as the
+      graphic, and we apply viewpoint rotations.
+     */
+    void connectToMapView(OverviewMapController* self, MapViewToolkit* view, Graphic* reticle)
+    {
+      QObject::connect(view, &MapViewToolkit::viewpointChanged, self,
+                       [self, view, reticle]
+                       {
+                         reticle->setGeometry(view->visibleArea());
+                         if (view->isNavigating())
+                         {
+                           const Viewpoint v = view->currentViewpoint(ViewpointType::CenterAndScale);
+                           Viewpoint newView{
+                               geometry_cast<Point>(v.targetGeometry()),
+                               v.targetScale() * self->scaleFactor(),
+                               v.rotation()};
+                           self->insetView()->setViewpoint(newView, 0);
+                         }
+                       });
+    }
+
+    /*!
+      \internal
+      Connect to SceneView's viewpointChanged. Note, we display centre and scale as the
+      graphic, and we ignore viewpoint rotations.
+     */
+    void connectToSceneView(OverviewMapController* self, SceneViewToolkit* view, Graphic* reticle)
+    {
+      QObject::connect(view, &SceneViewToolkit::viewpointChanged, self,
+                       [self, view, reticle]
+                       {
+                         const Viewpoint v = view->currentViewpoint(ViewpointType::CenterAndScale);
+                         reticle->setGeometry(v.targetGeometry());
+                         if (view->isNavigating())
+                         {
+                           Viewpoint newView{
+                               geometry_cast<Point>(v.targetGeometry()),
+                               v.targetScale() * self->scaleFactor()};
+                           self->insetView()->setViewpoint(newView, 0);
+                         }
+                       });
+    }
+  }
+
+  /*!
+    \internal
+    \brief If the inset view was navigated by the user, then this is called
+    to update the geoView viewpoint if it is a MapView.
+   */
+  void applyInsetNavigationToMapView(OverviewMapController* self, MapViewToolkit* view)
+  {
+    auto insetView = self->insetView();
+    const Viewpoint v = insetView->currentViewpoint(ViewpointType::CenterAndScale);
+    Viewpoint newView{
```

Comment:
Same recurring suggestion for these. I won't comment on them all.

```suggestion
    const Viewpoint newViewpoint{
```